eppo_client/eval.py

```diff
@@ -0,0 +1,62 @@
+from typing import Dict, Optional
+from eppo_client.sharding import Sharder
+from eppo_client.models import Range, Shard, Variation
+from eppo_client.rules import matches_rule
+import hashlib
+from dataclasses import dataclass
+import logging
+
+
+@dataclass
+class EvalResult:
+    flag_key: str
+    allocation_key: str
+    variation: Variation
+    extra_logging: Dict[str, str]
+    do_log: bool
+
+
+@dataclass
+class Evaluator:
+    sharder: Sharder
+
+    def evaluate_flag(
+        self, flag, subject_key, subject_attributes
```

Comment:
Also decided to pass along subject_key so we can include it in rule evaluation if we want (so you don't also have to add id to subject attributes), which has been a pet-peeve. We could just inject `"id": subject_key` into subject_attributes somewhere